src/store/CategoricalDisplayStore.js

```diff
@@ -195,7 +201,11 @@ export default class CategoricalDisplayStore {
             }
         }
 
-        return filteredItems;
+        if (searchHasResult) {
+            this.lastFilteredItems = filteredItems;
+        }
+
+        return this.lastFilteredItems;
```

Comment:
this may be confusing for people reading your code because it looks like you always return the last filtered items (even if you didn't). Try

```js
        if (searchHasResult) {
            this.lastFilteredItems = filteredItems;
            return filteredItems;
        } else {
            return this.lastFilteredItems;
       }
```